vmdb/app/models/vm_or_template/scanning.rb

```diff
@@ -183,11 +184,14 @@ def scan_metadata(category, options = {})
       "taskid" => nil,
       "vm_id" => self.id
     }.merge(options)
-    begin
-      run_command_via_parent("ScanMetadata", options)
-    rescue => err
-      $log.log_backtrace(err)
-    end
+    host = options.delete("host")
+    # If the options hash has an "args" element, remove it and add it to the "args" element with self.path
+    miqhost_args = Array(options.delete("args"))
+    options = {"args" => [self.path] + miqhost_args, "method_name" => "ScanMetadata", "vm_guid" => self.guid}.merge(options)
+    ost = OpenStruct.new(options)
+    host.call_ws(ost)
```

Comment:
@roliveri That is correct.  The `storage2proxies` method in vm_or_template.rb now only returns miq_server smartproxies instances which is the list used to selected the scanning proxy.